server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -112,12 +137,7 @@ public void define(WebService.NewController context) {
   public void handle(Request request, Response response) throws Exception {
     DbSession dbSession = dbClient.openSession(false);
     try {
-      SetRequest setRequest = toWsRequest(request);
-      Optional<ComponentDto> component = searchComponent(dbSession, setRequest);
-      checkPermissions(component);
-
-      validate(setRequest, component);
-      dbClient.propertiesDao().insertProperty(dbSession, toProperty(setRequest, component));
+      doHandle(dbSession, toWsRequest(request));
```

Comment:
It's not obvious for me that the validation on value, values, fieldValues is done in the toWsRequest.
I would move ``checkValueIsSet(setRequest);`` here